src/azure/azureController.ts

```diff
@@ -117,48 +117,61 @@ export class AzureController {
+	public async getAccount(accountStore: AccountStore): Promise<IAccount> {
 		let account: IAccount;
 		let config = vscode.workspace.getConfiguration('mssql').get('azureActiveDirectory');
 		if (config === utils.azureAuthTypeToString(AzureAuthType.AuthCodeGrant)) {
 			let azureCodeGrant = await this.createAuthCodeGrant();
 			account = await azureCodeGrant.startLogin();
 			await accountStore.addAccount(account);
-			if (!profile.tenantId) {
-				await this.promptForTenantChoice(account, profile);
-			}
-			let tid = profile.tenantId ? profile.tenantId : azureCodeGrant.getHomeTenant(account).id;
-			const token = await azureCodeGrant.getAccountSecurityToken(
-				account, tid, settings
-			);
-			if (!token) {
-				let errorMessage = LocalizedConstants.msgGetTokenFail;
-				this._vscodeWrapper.showErrorMessage(errorMessage);
-			}
-			profile.azureAccountToken = token.token;
-			profile.expiresOn = token.expiresOn;
-			profile.email = account.displayInfo.email;
-			profile.accountId = account.key.id;
 		} else if (config === utils.azureAuthTypeToString(AzureAuthType.DeviceCode)) {
 			let azureDeviceCode = await this.createDeviceCode();
 			account = await azureDeviceCode.startLogin();
 			await accountStore.addAccount(account);
-			if (!profile.tenantId) {
-				await this.promptForTenantChoice(account, profile);
-			}
-			let tid = profile.tenantId ? profile.tenantId : azureDeviceCode.getHomeTenant(account).id;
-			const token = await azureDeviceCode.getAccountSecurityToken(
-				account, tid, settings
+		}
+
+		return account;
+	}
+
+	public async getAccountSecurityToken(account: IAccount, tenantId: string | undefined, settings: AADResource): Promise<Token> {
+		let token;
+		let config = vscode.workspace.getConfiguration('mssql').get('azureActiveDirectory');
+		if (config === utils.azureAuthTypeToString(AzureAuthType.AuthCodeGrant)) {
+			let azureCodeGrant = await this.createAuthCodeGrant();
+			tenantId = tenantId ? tenantId : azureCodeGrant.getHomeTenant(account).id;
+			token = await azureCodeGrant.getAccountSecurityToken(
+				account, tenantId, settings
 			);
-			if (!token) {
-				let errorMessage = LocalizedConstants.msgGetTokenFail;
-				this._vscodeWrapper.showErrorMessage(errorMessage);
-			}
-			profile.azureAccountToken = token.token;
-			profile.expiresOn = token.expiresOn;
-			profile.email = account.displayInfo.email;
-			profile.accountId = account.key.id;
+		} else if (config === utils.azureAuthTypeToString(AzureAuthType.DeviceCode)) {
+			let azureDeviceCode = await this.createDeviceCode();
+			tenantId = tenantId ? tenantId : azureDeviceCode.getHomeTenant(account).id;
+			token = await azureDeviceCode.getAccountSecurityToken(
+				account, tenantId, settings
+			);
+		}
+		return token;
+	}
+
+	public async getTokens(profile: ConnectionProfile, accountStore: AccountStore, settings: AADResource): Promise<ConnectionProfile> {
```

Comment:
`populateAccountProperties` or something seems like a better name. Please also add doc comments for these new helper functions you're adding.